app/controllers/candidate_interface/application_feedback_controller.rb

```diff
@@ -0,0 +1,80 @@
+module CandidateInterface
+  class ApplicationFeedbackController < CandidateInterfaceController
+    def create
+      @application_feedback_form = CandidateInterface::ApplicationFeedbackForm.new(create_params)
+
+      if @application_feedback_form.save(current_application)
+        if @application_feedback_form.has_issues?
+          redirect_to candidate_interface_edit_application_feedback_path(
+            current_application.application_feedback.last.id,
+          )
+        else
+          redirect_to candidate_interface_application_feedback_thank_you_path
+        end
+      else
+        track_validation_error(@references_relationship_form)
+        flash[:warning] = add_errors_to_flash_message
+
+        redirect_to previous_path
+      end
+    end
+
+    def edit
+      @application_feedback = current_application.application_feedback.find(params[:id])
+      @application_feedback_form = CandidateInterface::ApplicationFeedbackForm.new
+    end
+
+    def update
+      @application_feedback = current_application.application_feedback.find(params[:id])
+      @application_feedback_form = CandidateInterface::ApplicationFeedbackForm.new(update_params)
+
+      if @application_feedback_form.update(@application_feedback)
+        redirect_to candidate_interface_application_feedback_thank_you_path
+      else
+        track_validation_error(@application_feedback_form)
+        render :edit
+      end
+    end
+
+    def thank_you
+      @application_feedback = current_application.application_feedback.last
+      @previous_path = if @application_feedback.id_in_path
+                         Rails.application.routes.url_helpers.send(@application_feedback.path, @application_feedback.id_in_path)
```

Comment:
There's probably a better way to do this than persist the related id to the DB, but this was the first solution. thought of. I did think about just passing the raw path in, but I like testing that we're passing valid path int the DB which this implementation allows us to do. Open to suggestions.

Reply:
We won't need this is the new design as users won't be redirected back to the page they came from.

Reply:
Agree it's nice to be able to validate the paths that get stored, does `ActionController::Routing::Routes.recognize_path` go some way to achieving this?